src/common/types/store-data/unified-data-interface.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// this is similar to `TestEngine` interface from axe-core
+export interface ScanEngineProperties {
+    name: string;
+    version: string;
+}
+
+export interface OSProperties {
+    name: string;
+    version?: string;
+    userAgent?: string;
+}
+
+export interface ViewPortProperties {
+    height?: number;
+    width?: number;
+}
+
+export interface PlatformData {
+    osInfo: OSProperties;
+    viewPortInfo: ViewPortProperties;
+}
+
+export interface ToolData {
+    scanEngineProperties: ScanEngineProperties;
+}
+
+export interface UnifiedResults {
+    results: UnifiedResult[];
+    platformInfo?: PlatformData;
+    toolInfo?: ToolData;
+}
+
+export interface UnifiedRule {
+    id: string;
+    description: string;
+}
+
+export interface InstancePropertyBag {
+    [property: string]: any;
+}
+
+export type StoredInstancePropertyBag = InstancePropertyBag;
+
+export interface UnifiedResult {
+    id: string;
+    status: ResultStatus;
+    evaluation: StoredInstancePropertyBag;
+    identifiers: StoredInstancePropertyBag;
+    descriptors: StoredInstancePropertyBag;
+    resolution: StoredInstancePropertyBag;
+}
+
+export type ResultStatus = 'pass' | 'fail' | 'unknown';
```

Comment:
I think pass/fail/unknown can work for our UI now (if we didn't need to map this to UI constructs we could just take a status at runtime) - it means anything that's not pass/fail from axe-android will map to unknown. axe-windows has warnings, for example. I think it makes sense to keep this list small initially and expand if needed.

Reply:
Thats the intention here, `unknown` is bigger in terms of what it can include; if we want to segregate this further; we can do it  (Expansion of these interfaces in future is the whole idea)